src/Common/tests/System/Net/Http/LoopbackServer.cs

```diff
@@ -786,17 +787,44 @@ public override async Task SendResponseAsync(HttpStatusCode statusCode = HttpSta
 
                 if (content != null || isFinal)
                 {
-                    headerString = GetHttpResponseHeaders(statusCode, headerString, contentLength, connectionClose : true);
+                    headerString = GetHttpResponseHeaders(statusCode, headerString, contentLength, connectionClose: true);
                 }
 
                 await SendResponseAsync(headerString).ConfigureAwait(false);
                 await SendResponseAsync(content).ConfigureAwait(false);
             }
 
+            public override async Task SendResponseHeadersAsync(HttpStatusCode statusCode = HttpStatusCode.OK, IList<HttpHeaderData> headers = null, int requestId = 0)
+            {
+                string headerString = null;
+
+                if (headers != null)
+                {
+                    foreach (HttpHeaderData headerData in headers)
+                    {
+                        headerString = headerString + $"{headerData.Name}: {headerData.Value}\r\n";
+                    }
+                }
+
+                headerString = GetHttpResponseHeaders(statusCode, headerString, 0, connectionClose: true);
+
+                await SendResponseAsync(headerString).ConfigureAwait(false);
+            }
+
             public override async Task SendResponseBodyAsync(byte[] body, bool isFinal = true, int requestId = 0)
             {
                 await SendResponseAsync(Encoding.UTF8.GetString(body)).ConfigureAwait(false);
             }
+
+            public override async Task WaitForCancellationAsync()
+            {
+                var buffer = new byte[1024];
+                try
+                {
+                    while (await ReadAsync(buffer, 0, buffer.Length).TimeoutAfter(TestHelper.PassingTestTimeoutMilliseconds).ConfigureAwait(false) > 0);
+                }
+                catch (IOException) { }
```

Comment:
Is it necessary to allow IOException here? I can imagine in some cases it could be, but not all. I worry a little we could be hiding some other issue. Probably not a big deal.

Reply:
Without it the tests I added fail, as the read ends up throwing an exception about the connection being closed abruptly.  I could however narrow it by checking the inner socket exception and validating it's a specific error code, if that helps.

Reply:
Although having said that, I can't repro the exception now.  Hmm.  I'll remove it if I continue to be unable to.